src/java.base/share/classes/jdk/internal/reflect/Reflection.java

```diff
@@ -56,7 +56,7 @@ public class Reflection {
         fieldFilterMap = Map.of(
             Reflection.class, ALL_MEMBERS,
             AccessibleObject.class, ALL_MEMBERS,
-            Class.class, Set.of("classLoader", "classData"),
+            Class.class, Set.of("classLoader", "classData", "protectionDomain"),
```

Comment:
Can you run a hello world with `-Xlog:class+init` to see if Reflection is initialized after `System$2` or something that implements JavaLangAccess?